optimum/exporters/onnx/convert.py

```diff
@@ -51,11 +51,15 @@ def check_dummy_inputs_are_allowed(
         model_inputs (`Iterable[str]`):
             The model input names.
     """
-    forward = (
-        model.forward
-        if is_torch_available() and issubclass(type(model), (PreTrainedModel, ModelMixin))
-        else model.call
-    )
+
+    classinfo = (PreTrainedModel,)
+    if is_diffusers_available():
+        classinfo += (ModelMixin,)
+
+    if "diffusers" in str(model.__class__) and not is_diffusers_available():
+        raise ImportError("The pip package `diffusers` is required to export stable diffusion models to ONNX.")
+
+    forward = model.forward if is_torch_available() and issubclass(type(model), classinfo) else model.call
```

Comment:
this seems very verbose and redundant, not sure if the check `issubclass` is necessary? @michaelbenayoun @echarlaix

Reply:
Yes, let's simplify this as you suggest!

Reply:
Should I just remove the check `issubclass(type(model), classinfo) ` ?